File: LeetCode/1137. N-th Tribonacci Number/solution.cpp
/**
 * 39 / 39 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.9 MB 
 */
class Solution {
public:
    int tribonacci(int n) {
        if (n < 2) return n;
        int a = 0, b = 1, c = 1, t1, t2;
        for (int i = 3; i <= n; ++ i) {
            t1 = c;
            t2 = b;
            c = a + b + c;
            b = t1;
            a = t2;
        }
        return c;
    }
};

File: LeetCode/1137. N-th Tribonacci Number/solution.py
"""
39 / 39 test cases passed.
Runtime: 44 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def tribonacci(self, n: int) -> int:
        if n < 2:
            return n;
        a, b, c = 0, 1, 1
        for i in range(3, n + 1):
            a, b, c = b, c, a + b + c
        return c
    
